controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,114 @@
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+)
+
+const (
+	providerAPIServerImage              = "PROVIDER_API_SERVER_IMAGE"
+	onboardingSecretGeneratorImage      = "ONBOARDING_SECRET_GENERATOR_IMAGE"
+	onboardingJobName                   = "onboarding-secret-generator"
+	onboardingTicketPublicKeySecretName = "onboarding-ticket-key"
+)
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("onboarding validation key generator job: Spec.AllowRemoteStorageConsumers is enabled")
+		if res, err := o.createJob(r, instance); err != nil {
+			return reconcile.Result{}, err
+		} else if !res.IsZero() {
+			return res, nil
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
+	err := r.Client.Delete(context.TODO(), onboardingSecretGeneratorJob)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.Info("Failed to delete resource", "Kind", onboardingSecretGeneratorJob.GetObjectKind(), "Name", onboardingSecretGeneratorJob.GetName())
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, err
+}
+
+func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingJobName,
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			// Eligible to delete automatically when job finishes
+			TTLSecondsAfterFinished: ptr.To(int32(0)),
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: onboardingJobName,
+					Containers: []corev1.Container{
+						{
+							Name:    onboardingJobName,
+							Image:   os.Getenv(onboardingSecretGeneratorImage),
+							Command: []string{"/usr/local/bin/onboarding-secret-generator"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  util.OperatorNamespaceEnvVar,
+									Value: os.Getenv(util.OperatorNamespaceEnvVar),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (o *onboardingValidationKeysGeneratorJob) createJob(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	var err error
+	if os.Getenv(onboardingSecretGeneratorImage) == "" {
+		err = fmt.Errorf("OnboardingSecretGeneratorImage env var is not set")
+		r.Log.Error(err, "No value set for env variable")
+
+		return reconcile.Result{}, err
+	}
+
+	actualSecret := &corev1.Secret{}
+	// Creating the job only if public is not found
+	err = r.Client.Get(context.Background(), types.NamespacedName{Name: onboardingTicketPublicKeySecretName,
```

Comment:
We should not create a new context for each client operation. 
Please use the context tracking the entire reconciliation. I believe it can be found in `r.ctx`